src/main/java/edu/usf/cutr/gtfs_realtime/bullrunner/GtfsRealtimeProviderImpl.java

```diff
@@ -537,99 +295,85 @@ private JSONArray downloadVehicleDetails(String external_route_id) throws IOExce
-            info.lat = (float) coordinate.getDouble("Latitude");
-            info.longi = (float) coordinate.getDouble("Longitude");
-            info.bearing = direction;
-
-            info.APCPercentage = child.getInt("APCPercentage");
+        BufferedReader reader = new BufferedReader(new InputStreamReader(in, "UTF-8"));
 
-            tripVehicleInfoMap.put(route, vehicleID, info);
+        StringBuilder builder = new StringBuilder();
+        String inputLine;
+        JSONArray response;
+        try {
+            while ((inputLine = reader.readLine()) != null)
+                builder.append(inputLine).append("\n");
+            response = new JSONArray(builder.toString());
+        } catch (SocketTimeoutException ex) {
+            _log.error("Error readline, server doesn't close the connection.", ex);
+            response = null;
         }
 
+        // keep only vehicles of the given route_id
+        JSONArray response_out = new JSONArray();
+        for (int i = 0; i < response.length(); i++){
+            JSONObject vehicleObj = response.getJSONObject(i);
+            String vehicle_id = vehicleObj.getString("id");
+            // check pattern name
+            URL URL_pattern = new URL(_url + "v1/vehicles/" + vehicle_id + "/arrivals?count=100&api-key=" + _api_key);
+            try {
+                connection = URL_pattern.openConnection();
+            } catch (Exception ex) {
+                _log.error("Error in opening feeds url", ex);
+            }
+            connection.setConnectTimeout(10000);  // connectTimeout is time out in miliseconds
+            connection.setReadTimeout(10000);
+            in = connection.getInputStream();
 
-    }
-
-    /**
-     * @return a JSON array parsed from the data pulled from the SEPTA vehicle
-     *         data API.
-     */
-    public class Pair {
-        private JSONArray array1;
-        private JSONArray array2;
-
-        public Pair(JSONArray array1, JSONArray array2) {
-            this.array1 = array1;
-            this.array2 = array2;
-
-        }
+            BufferedReader reader1 = new BufferedReader(new InputStreamReader(in, "UTF-8"));
 
-        public JSONArray getArray1() {
-            return array1;
+            StringBuilder builder1 = new StringBuilder();
+            String inputLine1;
+            JSONArray response_pattern;
+            try {
+                while ((inputLine1 = reader1.readLine()) != null)
+                    builder1.append(inputLine1).append("\n");
+                response_pattern = new JSONArray(builder1.toString());
+            } catch (SocketTimeoutException ex) {
+                _log.error("Error readline, server doesn't close the connection.", ex);
+                response_pattern = null;
+            }
+            String pattern_name = response_pattern.getJSONObject(1).getJSONObject("pattern").getString("name");
+
+            // check if pattern name matches
+            if (pattern_name.equals("Route C")){

```

Comment:
Let's move all of the following to `static final String` constants at the top of the method if all references are contained within this method, or the top of the file if they are used in more than one method:
* `Route C`
* `C`
* `MSC Express`

And, any other strings that we're trying to match in the API.